src/components/interface/SwitchTheme/SwitchTheme.js

```diff
@@ -39,19 +59,17 @@ const SwitchTheme = ({
       <ModalContent className="fr-form-group">
         <RadioGroup
           legend={legend}
-          value={currentTheme}
-          onChange={(value) => {
-            window.localStorage.setItem('prefers-color-scheme', value);
-            document.documentElement.setAttribute('data-fr-theme', value);
-          }}
+          value={window.localStorage.getItem('prefers-color-scheme')}
```

Comment:
this would make it work in SSR scenarios : `typeof window!=="undefined" && window.localStorage.getItem("prefers-color-scheme")`